### DISCUSSION TITLE
The configured Argon2 PHC in `ADMIN_TOKEN` is invalid: 'salt invalid: value to long

### DISCUSSION THREAD

**Discussion by janvgils** [dani-garcia/vaultwarden#3396] [Q&A] [answered]
When trying to switch to an argo2 ADMIN_TOKEN I ran into the following error and the vaultwarden server won't start. 

`The configured Argon2 PHC in ADMIN_TOKEN is invalid: 'salt invalid: value to long` 

I used the following configuration information and tried both the `/vaultwarden hash` and also the cli option. 

https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token.  

At the end I switched back to a random value `openssl rand -base64 48` and now everything is up and running again, aprt from the log message telling me the ADMIN_TOKEN is clear text and insecure.

Maybe someone can help me in the right direction and I am just overlooking something. 

Some details:

```
Running in a docker container.
env defined in a compose file (I already looked at the `$` escaping comments.
```
 
Thanks in advance.

**Comment by stefan0xC** [comment, marked as answer]
Please check again if you really have escaped all five occurrences of `$`. Because it sounds like you might have only escaped the first four.

At least I get the same error message if I simply remove the fifth `$`:
```
Error loading config:
  The configured Argon2 PHC in `ADMIN_TOKEN` is invalid: 'salt invalid: value to long'
```

**Comment by BlackDex** [comment]
The salt can't be larger then 128bits, which is 32 characters. The `vaultwarden hash` uses the maximum size already.
The password can be as large as you want 512 characters for example are not an issue.

And as @stefan0xC mentioned, make sure all the `$` are escaped with an extra `$`.

**Comment by janvgils** [comment]
Thank you all, at the end I missed one `$` to escape the string and then also ran into to `config.json` pit.

So after removing the `config.json` and making sure all `$` where escaped the server and web-vault are up and running.

It comes very accurate and is unforgiving.

Thank you all for the support.
